### PR TITLE
Command panel

### PR DESCRIPTION
**Pull request by liamkukid**
There is an implementation of todo list filtering, using localstorage to store data, and I tried to create my own hook.

### CODE REVIEW

File: src/Components/CommandPanel.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { FILTER_ACTIVE, FILTER_COMPLETED, SHOW_ALL } from '../Filters';
+
+import style from './commandPanel.module.scss';
+
+function CommandPanel({ todos, onFilterChanged, onRemoveCompleted }) {
+  return (
+    <div className={style.panel}>
+      <span className={style.span_items_left}>
+        {todos !== null &&
+          todos.length > 0 &&
```
Comment (aabrazouski) [inline src/Components/CommandPanel.js]: If array is empty we will have text ` items left`?

File: src/reducer.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+export const TODO_DONE_TYPE = 'TASK_DONE';
+export const ADD_TODO_TYPE = 'ADD_TODO';
+export const TODO_REMOVE_TYPE = 'TASK_REMOVE';
+export const RESET_TYPE = 'SET_UP';
+export const REMOVE_COMPLETED_TYPE = 'REMOVE_COMPLETED';
+
+const KEY_TODO = 'KEY_TODO';
+
+const reset = () => JSON.parse(localStorage.getItem(KEY_TODO));
```
Comment (aabrazouski) [inline src/reducer.js]: Why is it called `reset`?

File: src/reducer.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+const removeCompleted = todos => {
+  const newTodos = todos.filter(todo => !todo.done);
+  localStorage.setItem(KEY_TODO, JSON.stringify(newTodos));
+  return newTodos;
+};
+
+const addTodo = (todos, value) => {
+  const todo = { title: value, done: false, id: uuidv4() };
+  const newDotos = todos ? [...todos.slice(), todo] : [todo];
```
Comment (aabrazouski) [inline src/reducer.js]: Slice is redundant in this case. Actually, you can set default value as an empty array.

File: src/Components/CommandPanel.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      type="button"
+      onClick={onClick}
+      value={title}
+    >
+      {title}
+    </button>
+  );
+}
+
+export default memo(CommandPanel);
```
Comment (aabrazouski) [inline src/Components/CommandPanel.js]: Do not use `memo`. It is not necessary premature optimization.

File: src/Components/CommandPanel.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+      <Button
+        title="Completed"
+        onClick={() => onFilterChanged(FILTER_COMPLETED)}
+      />
+      <Button title="Remove Completed" onClick={onRemoveCompleted} />
+    </div>
+  );
+}
+
+function Button({ title, onClick }) {
```
Comment (aabrazouski) [inline src/Components/CommandPanel.js]: Move it to separate component, please

File: src/Components/CommandPanel.js (hunk 1 above)
Comment (liamkukid) [reply src/Components/CommandPanel.js]: fixed - it will be no items left

File: src/reducer.js (hunk 3 above)
Comment (liamkukid) [reply src/reducer.js]: fixed

File: src/Components/CommandPanel.js (hunk 4 above)
Comment (liamkukid) [reply src/Components/CommandPanel.js]: fixed

File: src/Components/CommandPanel.js (hunk 5 above)
Comment (liamkukid) [reply src/Components/CommandPanel.js]: fixed

**Pull request merged**